examples/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# =============================================================================
+# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+list(APPEND CMAKE_MESSAGE_CONTEXT "examples")
+
+if(MORPHEUS_PYTHON_INPLACE_BUILD)
+  link_directories(AFTER "${PROJECT_SOURCE_DIR}/morpheus/_lib")
+else()
+  # TODO: https://github.com/nv-morpheus/Morpheus/issues/331
+  link_directories(AFTER "${Python3_SITELIB}/morpheus/_lib")
```

Comment:
I will have to take a look at why you want to do this but we shouldnt be using `link_directories`. Everything should be using the `target_XXX` functions.

Reply:
In the `CMakeLists.txt` for both of the C++ examples I'm doing:
```
target_link_libraries(morpheus_example
    PUBLIC
      morpheus
      srf::pysrf
)
```

What ends up happening is the stage so file in the build dir is linked properly, however after its been installed `libmorpheus.so` can't be found.

If I recall correctly the way `libmorpheus.so` finds the others so's is because they are always in the same dir and we add `$ORIGIN` tp the RPATH:
`set(CMAKE_INSTALL_RPATH "$ORIGIN")`

and in the case of the unittests we are doing:
```
set_target_properties(test_libmorpheus
  PROPERTIES
    INSTALL_RPATH "$ORIGIN/.."
)
```

We could do the same here by setting `INSTALL_RPATH` but I think we would end up with a similar if statement inferring the location.